### PR TITLE
refactor: split test docs and add laravel support

### PR DESCRIPTION
**Pull request by vinceAmstoutz**
As discussed in our messages with @dunglas, I will take the following steps:

- [x] Create a Documentation File for Laravel: I will create a new document titled `laravel/testing.md` that describes how to write tests using `Pest` and `PHPUnit` in `Laravel`.
- [x] Reorganize Symfony Documentation: To improve separation of concerns, I will move the Symfony-related content from `core/testing.md` into `symfony/testing.md.`
- [x] Maintain Core Documentation: The `core/testing.md` file will remain the central resource for testing documentation encompassing both `Symfony` and `Laravel`.

### CODE REVIEW

**Comment by vinceAmstoutz** [comment]
> Maybe can you keep the root `core/testing.md` file, move most of its content in `symfony/testing.md` directly (no need for two files IMHO), add a new `laravel/testing.md` file, and reference the Laravel and Symfony testing docs in `docs/testing.md`.
> 
> This will prevent setting redirections.

Indeed, that's better, I'll do that!

File: laravel/testing.md [hunk 1]
```diff
@@ -0,0 +1,520 @@
+# Testing the API with Laravel
+
+For an introduction to testing using API Platform, refer to the [Core Testing Documentation](/core/testing.md), or access the
```
Comment (dunglas) [inline laravel/testing.md:3]:
```suggestion
For an introduction to testing using API Platform, refer to the [Core Testing Documentation](../core/testing.md), or access the
```

File: laravel/testing.md [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+[Symfony Testing Guide](/symfony/testing.md).
```
Comment (dunglas) [inline laravel/testing.md:4]:
```suggestion
[Symfony Testing Guide](../symfony/testing.md).
```

File: laravel/testing.md [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+> with PHPUnit's assertion API or if you need to perform more complex assertions that aren't available in Pest's expectation API.
+> For more information see the [Pest Assertion API](https://pestphp.com/docs/writing-tests#content-assertion-api) documentation.
+
+### Installing Pest
+
+By default, when using Laravel, Pest is pre-configured through the Composer plugin `pestphp/pest-plugin`. You can find this plugin listed in the `allow-plugins` section of your `composer.json` file.
+
+To check the Pest installation, run the following command:
+
+```bash
```
Comment (dunglas) [inline laravel/testing.md:32]:
```suggestion
```console
```

Same everywhere

File: laravel/testing.md [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+php artisan test
+```
+
+If for some reason, Pest is not installed refer to the [Pest Installation Guide](https://pestphp.com/docs/installation).
+
+In that case, you can run Pest using:
+```bash
```
Comment (dunglas) [inline laravel/testing.md:39]:
```suggestion
In that case, you can run Pest using:

```console
```

File: laravel/testing.md [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+### Writing Functional Tests with Pest
+
+#### Generate the Factory
+
+Using Laravel, you can efficiently test databases by combining seeding with model factories. Model factories allow you
+to generate large amounts of test data quickly, while seeding ensures your database is pre-populated with the necessary records.
+
+To create a factory for your model, you can use [Laravel Artisan](https://laravel.com/docs/artisan) command.
+For example, to create a factory for a Book model, run:
+```bash
```
Comment (dunglas) [inline laravel/testing.md:52]:
```suggestion
For example, to create a factory for a Book model, run:

```console
```

File: laravel/testing.md [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+
+The API Platform distribution is [shipped with a GitHub Actions workflow](https://github.com/api-platform/api-platform/blob/main/.github/workflows/ci.yml) that builds the Docker images, does a [smoke test](<https://en.wikipedia.org/wiki/Smoke_testing_(software)>) to check that the application's entrypoint is accessible, and runs PHPUnit.
+
+The API Platform Demo [contains a CD workflow](https://github.com/api-platform/demo/tree/main/.github/workflows) that uses [the Helm chart provided with the distribution](../deployment/kubernetes.md) to deploy the app on a Kubernetes cluster.
+
+## Additional and Alternative Testing Tools
+
+You may also be interested in these alternative testing tools (not included in the API Platform distribution):
+
+- [Hoppscotch](https://docs.hoppscotch.io/features/tests), create functional test for your API
```
Comment (dunglas) [inline laravel/testing.md:451]: Hoppstoch is also on the next line

File: laravel/testing.md [hunk 7]
```diff
@@ -0,0 +1,520 @@
...
+- [Hoppscotch](https://docs.hoppscotch.io/documentation/features/rest-api-testing/), create functional test for your API
+  Platform project using a nice UI, benefit from its Swagger integration and run tests in the CI using [the command-line tool](https://docs.hoppscotch.io/cli);
+- [Behat](https://behat.org), a [behavior-driven development (BDD)](https://en.wikipedia.org/wiki/Behavior-driven_development) framework to write the API specification as user
+  stories and in natural language then execute these scenarios against the application to validate its behavior;
+- [Blackfire Player](https://blackfire.io/player), a nice DSL to crawl HTTP services, assert responses, and extract data
+  from HTML/XML/JSON responses;
+- [PHP Matcher](https://github.com/coduo/php-matcher), the Swiss Army knife of JSON document testing.
```
Comment (dunglas) [inline laravel/testing.md:458]: You can remove these two ones imho

File: laravel/testing.md [hunk 8]
```diff
@@ -0,0 +1,520 @@
...
+- [Playwright](https://playwright.dev) is recommended if you use have PWA/JavaScript-heavy app.
+
+## Testing Utilities for Laravel
+
+### API Test Assertions with Laravel
+
+In addition to [the built-in ones](https://phpunit.readthedocs.io/en/11.4/assertions.html), API Platform provides
```
Comment (dunglas) [inline laravel/testing.md:465]:
```suggestion
In addition to [the built-in ones](https://phpunit.readthedocs.io/en/main/assertions.html), API Platform provides
```

File: symfony/testing.md [hunk 9]
```diff
@@ -1,11 +1,9 @@
-# Testing the API
+# Testing the API with Symfony
 
-Now that you have a functional API, you should write tests to ensure it has no bugs, and to prevent future regressions.
-Some would argue that it's even better to [write tests first](https://martinfowler.com/bliki/TestDrivenDevelopment.html).
+For an introduction to testing using API Platform, refer to the [Core Testing Documentation](/core/testing.md), or access the
+[Laravel Testing Guide](/laravel/testing.md).
```
Comment (dunglas) [inline symfony/testing.md:4]:
```suggestion
For an introduction to testing using API Platform, refer to the [Core Testing Documentation](../core/testing.md), or access the
[Laravel Testing Guide](../laravel/testing.md).
```

File: symfony/testing.md [hunk 10]
```diff
@@ -374,13 +364,13 @@ You may also be interested in these alternative testing tools (not included in t
...
-  [behavior-driven development (BDD)](https://en.wikipedia.org/wiki/Behavior-driven_development) framework to write the API
-  specification as user stories and in natural language then execute these scenarios against the application to validate
-  its behavior;
+- [Behat](https://behat.org), a [behavior-driven development (BDD)](https://en.wikipedia.org/wiki/Behavior-driven_development) framework to write the API specification as user
+  stories and in natural language then execute these scenarios against the application to validate its behavior;
 - [Blackfire Player](https://blackfire.io/player), a nice DSL to crawl HTTP services, assert responses, and extract data
   from HTML/XML/JSON responses;
 - [PHP Matcher](https://github.com/coduo/php-matcher), the Swiss Army knife of JSON document testing.
+- [Pest](https://pestphp.com/) is a testing framework built on top of [PHPUnit](https://phpunit.de/) and introduces
+  additional features along with a syntax inspired by Ruby's RSpec and the Jest testing APIs.
```
Comment (dunglas) [inline symfony/testing.md:373]: You can remove these ones imho (Pest is not popular in the Symfony ecosystem)

File: laravel/testing.md (hunk 6 above)
Comment (vinceAmstoutz) [reply laravel/testing.md:453]: This was already the case but I deleted the 2nd one.

**Review by dunglas — APPROVED** [review]

**Pull request merged**